model_table_struct.go

```diff
@@ -74,15 +74,11 @@ func (m *structTableModel) initStruct() error {
 		return m.structInitErr
 	case reflect.Interface:
 		m.strct = m.strct.Elem()
-	}
-
-	if m.strct.Kind() == reflect.Ptr {
+	case reflect.Ptr:
```

Comment:
This is not safe, because on line 76 we do `m.strct = m.strct.Elem()` which might return a nil ptr.